### ISSUE TITLE
Incorrect refactor in async example

### ISSUE THREAD

**Issue by jrobichaud** [github/eslint-plugin-github#133]
About this example:
https://github.com/github/eslint-plugin-github/blob/a8ec7ad42ce6111df6d8be08abf50a3a455d85c2/docs/rules/array-foreach.md#L83-L100

The following code creates N async promises in parallel and wait until they are all completed (it polishes all apples at the same time until they are all polished):

```javascript
await Promise.all( 
    apples.map(polishApple) 
) 
```

The following code awaits for each async call individually (it polishes 1 apple at time until they are all polished).

```javascript
for (const apple of apples) { 
    await polishApple(apple) 
} 
```

This is a refactor changing substantially the behaviour of the code.

**Comment by koddsson** [comment]
I think the docs aren't suggesting that you should refactor code example A to code example B but is rather trying to showcase the difference between the two examples when it comes to refactoring existing code.

We still might want to make the example a bit better so that there isn't any misunderstanding.
